lib/resty/dns/resolver.lua

```diff
@@ -100,7 +100,8 @@ for i = 2, 64, 2 do
 end
 
 
-function _M.new(class, opts)
+function _M.new(opts_dot, opts)
+    opts = opts or (opts_dot ~= _M and opts_dot) or nil
```

Comment:
I don't want this since it's too fragile and complicated though I regret accepting the class argument as the first parameter. This line will have problems when we add a 3rd parameter in the future.